test/utils/api.js

```diff
@@ -0,0 +1,150 @@
+		});
+
+		it('should have mainnet nethash', () => {
+			return expect(apiClient.headers.nethash).to.be.equal(mainnetNethash);
+		});
+
+		it('should have currentNode as first element of the api.nodes', () => {
+			return expect(apiClient.currentNode).to.be.equal(apiConfig.nodes[0]);
+		});
+	});
+
+	describe('when the network is set to main and nodes are empty', () => {
+		beforeEach(() => {
+			apiConfig = {
+				network: 'main',
+				nodes: [],
+			};
+			apiClient = getAPIClient(apiConfig);
+			return Promise.resolve();
+		});
+
+		it('should have mainnet nethash', () => {
+			return expect(apiClient.headers.nethash).to.be.equal(mainnetNethash);
+		});
+
+		it('should have currentNode as the mainnet node', () => {
+			return expect(mainnetNodes).to.include(apiClient.currentNode);
+		});
+	});
+
+	describe('when the network is set to test and nodes are provided', () => {
+		beforeEach(() => {
+			apiConfig = {
+				network: 'test',
+				nodes: ['http://localhost:4000'],
+			};
+			apiClient = getAPIClient(apiConfig);
+			return Promise.resolve();
+		});
+
+		it('should have testnet nethash', () => {
+			return expect(apiClient.headers.nethash).to.be.equal(testnetNethash);
+		});
+
+		it('should have currentNode as first element of the api.nodes', () => {
+			return expect(apiClient.currentNode).to.be.equal(apiConfig.nodes[0]);
+		});
+	});
+
+	describe('when the network is set to test and the nodes are empty', () => {
+		beforeEach(() => {
+			apiConfig = {
+				network: 'test',
+				nodes: [],
+			};
+			apiClient = getAPIClient(apiConfig);
+			return Promise.resolve();
+		});
+
+		it('should have mainnet nethash', () => {
+			return expect(apiClient.headers.nethash).to.be.equal(testnetNethash);
+		});
+
+		it('should have currentNode as the default testnet node', () => {
+			return expect(apiClient.currentNode).to.be.equal(testnetNode);
+		});
+	});
+
+	describe('when the network is set to nethash and nodes are provided', () => {
+		beforeEach(() => {
+			apiConfig = {
+				network:
+					'ef3844327d1fd0fc5aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa2b7e859e9ca0c',
+				nodes: ['http://localhost:4000'],
+			};
+			apiClient = getAPIClient(apiConfig);
+			return Promise.resolve();
+		});
+
+		it('should have the custom nethash', () => {
+			return expect(apiClient.headers.nethash).to.be.equal(apiConfig.network);
+		});
+
+		it('should have currentNode as first element of the api.nodes', () => {
+			return expect(apiClient.currentNode).to.be.equal(apiConfig.nodes[0]);
+		});
+	});
+
+	describe('when the network is set to custom nethash and nodes are empty', () => {
+		beforeEach(() => {
+			apiConfig = {
+				network:
+					'ef3844327d1fd0fc5aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa2b7e859e9ca0c',
+				nodes: [],
+			};
+			return Promise.resolve();
+		});
+
+		it('should throw an error', () => {
+			return expect(getAPIClient.bind(null, apiConfig)).to.throw(
+				'APIClient requires nodes for initialization.',
+			);
+		});
+	});
```

Comment:
What about if `apiConfig.network` is not a valid nethash?

Reply:
This was previously checked at the command level as invalid input, so there is not check on the source code too.
Also, `elements` any inputs too, but it is not used other than header ='(
maybe we disallow to set the custom nethash coz it's not useful in anyway?